[17.0][MIG] stock_barcodes

edescalona opened the pull request:
@BinhexTeam

This module migration was based on PR #668 

Depends on:

- https://github.com/OCA/stock-logistics-workflow/pull/2132

sergio-teruel committed:
[ADD] stock_barcodes: New module to read barcodes on stocks operations (#195)

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update stock_barcodes.pot

astirpe committed:
[12.0][MIG] stock_barcodes

sergio-teruel committed:
[ADD] stock_barcodes: Allow to read barcodes from pickings

pedrobaeza committed:
[FIX] stock_barcodes: Fix ES translation

sergio-teruel committed:
[ADD] stock_barcodes: Allow to read barcodes from pickings. Last changes

sergio-teruel committed:
[FIX] stock_barcodes: Fix validate picking from candidate picking kanban view

sergio-teruel committed:
[IMP] stock_barcodes: Hidden lot for non tracking products.
When user creates a lot, set default product the wizard selected product.
When confirm the lot creation wizard, set this lot in lot wizard scanning barcode field.

oca-travis committed:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-11.0/stock-logistics-barcode-11.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-11-0/stock-logistics-barcode-11-0-stock_barcodes/

oca-travis committed:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-12.0/stock-logistics-barcode-12.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-12-0/stock-logistics-barcode-12-0-stock_barcodes/

chienandalu committed:
[FIX] stock_barcodes: out picking location workflow

When we've got an out picking, the logical workflow is that the scanned location
is the location we're getting the stock from

oca-travis committed:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
stock_barcodes 12.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-12.0/stock-logistics-barcode-12.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-12-0/stock-logistics-barcode-12-0-stock_barcodes/

chienandalu committed:
[FIX] stock_barcodes: location_dest_id

OCA-git-bot committed:
stock_barcodes 12.0.1.1.1

sergio-teruel committed:
[IMP] stock_barcodes: black, isort

sergio-teruel committed:
[MIG] stock_barcodes: Migration to v13.0

oca-travis committed:
[UPD] Update stock_barcodes.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-13.0/stock-logistics-barcode-13.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-13-0/stock-logistics-barcode-13-0-stock_barcodes/

pedrobaeza committed:
[FIX] stock_barcodes: Make tests more resilient

UID = 1 is sometimes called "System" and sometimes "OdooBot", depending on the
installed set of modules, so tests can fail if you rely on this data.

Removing that variable from the tests, we get resilient tests.

pedrobaeza committed:
[FIX] stock_barcodes: Another non resilient test

Followup of dab52c9. There's another test that hardcodes OdooBot name for uid=1.

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-13.0/stock-logistics-barcode-13.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-13-0/stock-logistics-barcode-13-0-stock_barcodes/

sergio-teruel committed:
[IMP] stock_barcodes: Add barcode field to location view in same group like as v12.0

oca-travis committed:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
stock_barcodes 13.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-13.0/stock-logistics-barcode-13.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-13-0/stock-logistics-barcode-13-0-stock_barcodes/

chienandalu committed:
[IMP] stock_barcodes: allow to add lines even if not qtys reserved

If we had any reservation (i.e unreserve), the barcodes button was
hidden.

sergio-teruel committed:
[ADD] stock_barcodes: Add option to get lots automatically in inventory depends on removal strategy
TT27844

oca-travis committed:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
stock_barcodes 13.0.1.2.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-13.0/stock-logistics-barcode-13.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-13-0/stock-logistics-barcode-13-0-stock_barcodes/

fshah-initos committed:
[IMP] stock_barcodes: black, isort, prettier

fshah-initos committed:
[14.0][MIG]stock_barcodes

etobella committed:
[FIX] stock_barcodes: use the uom of the package if defined

etobella committed:
[IMP] stock_barcodes: Apply putaway strategy

chienandalu committed:
[FIX] stock_barcodes: avoid mixing tracked and untracked quants

Using the auto lot feature in the inventory scanning we want to avoid
distributing quantities to quants with no lot if the products if the
product change from untracked to tracked.

TT31267

chienandalu committed:
[FIX] stock_barcodes: reset lot id if we change the product manually

Otherwise the previous product lot would persist

TT31267

oca-travis committed:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[UPD] README.rst

sergio-teruel committed:
[IMP] stock_barcodes: Refactoring code and improvements

OCA-git-bot committed:
stock_barcodes 14.0.1.1.0

mariadforgeflow committed:
[FIX] stock_barcodes: make available the action to read barcodes

OCA-git-bot committed:
stock_barcodes 14.0.1.2.0

LoisRForgeFlow committed:
[FIX] stock_barcodes: take company from env not from user

OCA-git-bot committed:
stock_barcodes 14.0.1.2.1

LoisRForgeFlow committed:
[FIX] stock_barcodes: read act windows permission issue

OCA-git-bot committed:
stock_barcodes 14.0.1.2.2

LoisRForgeFlow committed:
[FIX] stock_barcodes: set option to create lot to true in test

It was de default behavior in GS1 before the refactor

LoisRForgeFlow committed:
[FIX] stock_barcodes: several small adjustments.

* When a picking is processed partially it was not
  posible to validate it because the backorder wizard
  was captured and never returned to the user.
* Only update sml locations when there is a location
  to update, otherwise constraints for non null values
  could raise.
* Invalidate cache of todo lines when determining
  next action to do to ensure that line status is
  updated.
* Reword string in button from "Force" to "Done"
  as it was a bit misleading. Actually, in spanish
  translation it was already being translated to
  "Hecho" ("Done").

LoisRForgeFlow committed:
[IMP] stock_barcodes: show quantity even when not editable.

LoisRForgeFlow committed:
[MIG] stock_barcodes: clean warnings and adapt to v14.

Adapt last refector done in v13 to v14.

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
stock_barcodes 14.0.2.0.0

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-14.0/stock-logistics-barcode-14.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-14-0/stock-logistics-barcode-14-0-stock_barcodes/

francesco-ooops committed:
Added translation using Weblate (Italian)

SicurSam committed:
Translated using Weblate (Italian)

Currently translated at 35.6% (94 of 264 strings)

Translation: stock-logistics-barcode-14.0/stock-logistics-barcode-14.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-14-0/stock-logistics-barcode-14-0-stock_barcodes/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 35.9% (95 of 264 strings)

Translation: stock-logistics-barcode-14.0/stock-logistics-barcode-14.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-14-0/stock-logistics-barcode-14-0-stock_barcodes/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (264 of 264 strings)

Translation: stock-logistics-barcode-14.0/stock-logistics-barcode-14.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-14-0/stock-logistics-barcode-14-0-stock_barcodes/it/

LoisRForgeFlow committed:
[FIX] stock_barcodes: make actions readable by non-admin users.

leftover from migration.

LoisRForgeFlow committed:
[FIX] stock_barcodes: do not use PO UoM by default

When creating new stock move lines, we should use the UoM in the
move, if any, or the default UoM of the product.

OCA-git-bot committed:
stock_barcodes 14.0.2.0.1

LoisRForgeFlow committed:
[FIX] stock_barcodes: remove last scan do not revert last action.

In pickings, the scan log was always generated without details of
the move lines modified/created. Without this information in the log
removing the log was dong nothing else than just that, removing the
log; leaving the last operation still reflected in the picking.

Also, when the picking was a draft one, the stock move line and
the stock move could have been created by the scan itself, so we
need to remove it in that case, reverting the action would leave
the stock move there that could be validated by mistake.

LoisRForgeFlow committed:
[FIX] stock_barcodes: use float_compare to avoid rounding issues

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
stock_barcodes 14.0.2.0.2

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-14.0/stock-logistics-barcode-14.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-14-0/stock-logistics-barcode-14-0-stock_barcodes/

sergio-teruel committed:
[IMP] stock_barcodes: Clean wizard values after the move to do has been reset

OCA-git-bot committed:
stock_barcodes 14.0.2.0.3

etobella committed:
[IMP] stock_barcodes: black, isort, prettier

etobella committed:
[MIG] stock_barcodes: Migration to 15.0

mariadforgeflow committed:
[IMP] stock_barcodes: add Barcode Option Group for not planned pickings

LoisRForgeFlow committed:
[MIG/FIX] stock_barcodes: fix notification usage to adapt to v15.

LoisRForgeFlow committed:
[IMP] stock_barcodes: attend new pylint mandatory checks

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
stock_barcodes 15.0.1.1.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

MiquelRForgeFlow committed:
[FIX] stock_barcodes: add migration script

sergio-teruel committed:
[FIX] stock_barcodes: Do not create a lot with create lot option enabled for products without tracking

sergio-teruel committed:
[FIX] stock_barcodes: Pending moves are not displayed when moves origin is stock_moves without reservations

OCA-git-bot committed:
stock_barcodes 15.0.1.2.0

sergio-teruel committed:
[FIX] stock_barcodes: put in pack call method fails

OCA-git-bot committed:
stock_barcodes 15.0.1.2.1

sergio-teruel committed:
[IMP] stock_barcodes: take into account package fields when scan barcodes

sergio-teruel committed:
[FIX] stock_barcodes: Not pending moves when origin pending moves is stock move lines
TT41789

OCA-git-bot committed:
stock_barcodes 15.0.1.2.2

sergio-teruel committed:
[IMP] stock_barcodes: Add prepare method to set wizard values from picking to be more inheritable

pere-aquarian committed:
Translated using Weblate (Spanish)

Currently translated at 69.1% (175 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Allow display detailed operations in barcode wizard

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Display partner on wizard

sergio-teruel committed:
[IMP] stock_barcodes: Do not clean values forced in an action done

sergio-teruel committed:
[IMP] stock_barcodes: add package to moves todo group key

sergio-teruel committed:
[IMP] stock_barcodes: display picking notes in wizard barcode

sergio-teruel committed:
[IMP] stock_barcodes: display improvements

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

carlosdauden committed:
[IMP] stock_barcodes: Show same type ready pickings tree after validate picking

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Allow make inventories

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[FIX] stock_barcodes: Check serial product tracking for inventory when a quant is created

sergio-teruel committed:
[IMP] stock_barcodes: Improve heritability of write on stock move lines

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

OCA-git-bot committed:
[UPD] README.rst

sergio-teruel committed:
[IMP] stock_barcodes: Better inheritance of qty focus

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Add hook to process values when moves todo have been created.

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Test improvements and create lots in confirm action

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[BOT] post-merge updates

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (253 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/es/

Commit:
[UPD] Update stock_barcodes.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 90.2% (249 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (276 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/es/

sergio-teruel committed:
[IMP] stock_barcodes: Filled all fields if they is defined in options as filled default.
Determine scan step after determine todo action

sergio-teruel committed:
[FIX] stock_barcodes: Incorrect order of detailed operations fixed

mymage committed:
Translated using Weblate (Italian)

Currently translated at 90.2% (249 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] Add security group to barcodes main menu

sergio-teruel committed:
[IMP] stock_barcodes: Allow skip clean values when a sml is created or updated in action_done method.

sergio-teruel committed:
[IMP] stock_barcodes: Filter sml lines bby complete packages

sergio-teruel committed:
[IMP] stock_barcodes: Display only the quants pending to read. Allow set to zero

mymage committed:
Translated using Weblate (Italian)

Currently translated at 90.2% (249 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Add option to keep wizard values until the pending move is completed

mymage committed:
Translated using Weblate (Italian)

Currently translated at 90.2% (249 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Try fill wizard data after product is read if fill_fields_from_lot is checked

sergio-teruel committed:
[IMP] stock_barcodes: Add total product qty without lot, package, etc

sergio-teruel committed:
[IMP] stock_barcodes: Set order in inventory quants

CarlosRoca13 committed:
[IMP] stock_barcodes: Visual improvements on footer

sergio-teruel committed:
[FIX] stock_barcodes: Reload todo action after unlink sml done

sergio-teruel committed:
[IMP] stock_barcodes Set to readonly total qty fields

sergio-teruel committed:
[IMP] stock_barcodes: Recompute fill_todo_records after unlink sml.

sergio-teruel committed:
[IMP] stock_barcodes: make todo qty_done compute readonly to better computation of pending lines

CarlosRoca13 committed:
[IMP] stock_barcodes: Responsive dropdown

sergio-teruel committed:
[IMP] stock_barcodes: Return stock move line created or updated after confirm action

carlosdauden committed:
[IMP] stock_barcodes: Ignore rest instead of Done to avoid that user skip pending operations

mymage committed:
Translated using Weblate (Italian)

Currently translated at 90.2% (249 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Inject a context to know when a picking is validated from barcodes interface

sergio-teruel committed:
[FIX] stock_barcodes: Set incorrect package and owner in set info from quants

sergio-teruel committed:
[FIX] stock_barcodes: create new lines when qty_done is greather than initial demand
TT46503

sergio-teruel committed:
[FIX] stock_barcodes: qty_done is increased if user scan a package twice

sergio-teruel committed:
[FIX] stock_barcodes: Link to correct move whe a sml is created

sergio-teruel committed:
[FIX] stock_barcodes: Apply locations domain in pickings and inventory

sergio-teruel committed:
[FIX] stock_barcodes: Always re-write las location scanned is not candidate sml's was found

sergio-teruel committed:
[IMP] stock_barcodes: Add context to avoid update quantities from lot if fill data from lot is active

sergio-teruel committed:
[IMP] stock_barcodes: Search packages in all internal locations is forced option is not checked

sergio-teruel committed:
[FIX] stock_barcodes: The package is not taken into account when user does an inventory

carlosdauden committed:
[IMP] stock_barcodes: Add accumulate_read_quantity option to be used in inventory. In the future it can be used by other process.
TT46934

carlosdauden committed:
[IMP] stock_barcodes: Allow find package in external locations. Add barcode_domain_field context
TT46823

carlosdauden committed:
[IMP] stock_barcodes: Fill inventory quantity when scan package
TT46934

mymage committed:
Translated using Weblate (Italian)

Currently translated at 90.2% (249 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

carlosdauden committed:
[IMP] stock_barcodes: Add option ignore_quant_location
TT47469

sergio-teruel committed:
[FIX] stock_barcodes: product not take into account to search quants.

sergio-teruel committed:
[IMP] stock_barcodes: Limit product quantities entry to avoid barcode read errors over an input element
TT48037

FranzPoize committed:
[MIG] stock_barcodes: Migration to 16.0

sergio-teruel committed:
[IMP] stock_barcodes: Allow prefill product_qty when not in manual entry mode
TT48077

sergio-teruel committed:
[IMP] stock_barcodes: Add context to skip set info from quants although the configuration option is activated

sergio-teruel committed:
[IMP] stock_barcodes: Allow display notifications to web client

sergio-teruel committed:
[IMP] stock_barcodes: Allow apply putaway strategy as screen option

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

sergio-teruel committed:
[FIX] stock_barcodes: Required lot_id although lot_name is set without create lot option is checked.
Display notification for required field.

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

sergio-teruel committed:
[IMP] stock_barcodes: Skip backorder creation for pickings with all moves read or ignore pending quantities

sergio-teruel committed:
[IMP] stock_barcodes: Add option group to allow to select if you want order by location or location_dest_id

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

chienandalu committed:
[MIG] stock_barcodes: followup migration

TT44219

FranzPoize committed:
[MIG] stock_barcodes: more migrating

sergio-teruel committed:
[FIX] stock_barcodes: Do not search quant when a only lot is scanned
TT49023

chienandalu committed:
[MIG] stock_barcodes: continue migration

- Drop scan read logs feature, which use was deprecated already in the
  previous versions.
- Pass bootstrap migration scripts.
- Fix views issues.
- Other code fixes.

sergio-teruel committed:
[IMP] stock_barcodes: Refresh data. Change dependencies and call bus to reload wizard data

sergio-teruel committed:
[FIX] stock_barcodes: todo lines not refresh data for automatic mode when a new stock move line is created

sergio-teruel committed:
[FIX] stock_barcodes: The field state from stock move line compared is not correct
TT49399

sergio-teruel committed:
[FIX] stock_barcodes: Notify events not managed by resource

sergio-teruel committed:
[FIX] stock_barcodes: Crash if user defines an option to scan with a field which not exists

chienandalu committed:
[FIX] stock_barcodes: styling

alexis-via committed:
[IMP] stock_barcodes: inherit create in batch to avoid warning

carlosdauden committed:
[IMP] stock_barcodes: Add pending move with difference between demand and reserved

carlosdauden committed:
[IMP] stock_barcodes: Performance and extra checks from 16.0

carlosdauden committed:
[IMP] stock_barcodes: Avoid NewId environment calling method on_barcode_scanned instead of onchange
Onchange limits documented in method:
It is not possible for a ``one2many`` or ``many2many`` field to modify
itself via onchange. This is a webclient limitation - see `#2693 <https://github.com/odoo/odoo/issues/2693>`_.

This causes refresh problems when creating or modifying records, and it is also not ideal to create records in "onchange mode"

carlosdauden committed:
[FIX] stock_barcodes: Remove dead code

sergio-teruel committed:
[FIX] stock_barcodes: Fix tests

sergio-teruel committed:
[FIX] stock_barcodes: Move qty_available field from read base to read picking

sergio-teruel committed:
[IMP] stock_barcodes: Add display read items on inventory mode in option group and minor fix

Commit:
[UPD] Update stock_barcodes.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-stock_barcodes/

houzefa-abba committed:
Added translation using Weblate (French)

houzefa-abba committed:
Translated using Weblate (French)

Currently translated at 69.1% (208 of 301 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-stock_barcodes/fr/

houzefa-abba committed:
Translated using Weblate (French)

Currently translated at 76.4% (230 of 301 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-stock_barcodes/fr/

houzefa-abba committed:
Translated using Weblate (French)

Currently translated at 77.7% (234 of 301 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-stock_barcodes/fr/

alexis-via committed:
[FIX] stock_barcodes: play error sound when requested

OCA-git-bot committed:
[BOT] post-merge updates

carlosdauden committed:
[FIX] stock_barcodes: Quant and stock move line reserve are unsynchronized after button "no create backorder".
TT51871

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (301 of 301 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-stock_barcodes/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (301 of 301 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-stock_barcodes/it/

alexis-via committed:
[IMP] stock_barcodes: use trash icons instead of recycle icons in inventory frontend

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update stock_barcodes.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-stock_barcodes/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (301 of 301 strings)

Translation: stock-logistics-barcode-16.0/stock-logistics-barcode-16.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-16-0/stock-logistics-barcode-16-0-stock_barcodes/it/

edescalona committed:
[16.0][IMP] stock_barcodes: Visual and functional improvements to the Inventory and Operations scan views

[16.0][IMP] stock_barcodes: Updating menu with the new action.

[16.0][IMP] stock_barcodes: If the required quantity of the product is exceeded, allow the user to decide on the excess quantity.

[16.0][IMP] stock_barcodes: Change format string

[16.0][IMP] stock_barcodes: Readme

[16.0][IMP] stock_barcodes: Readme

[16.0][FIX] stock_barcodes: Validating existing method

[16.0][IMP] stock_barcodes: Changing the format of f-strings (f) to {}.format()

[16.0][IMP] stock_barcodes: Add bucle for in constrains. Add return domain

[16.0][IMP] stock_barcodes: Changing ID of barcode main action

[16.0][IMP] stock_barcodes: Add tests.

[16.0][IMP] stock_barcodes: Showing the pickings of the same type as the one previously validated.

[16.0][IMP] stock_barcodes: Updating menu with the new action.

[16.0][IMP] stock_barcodes: Add styles button confirm force done.

edescalona commented:
Ready for review

edescalona committed:
[IMP] stock_barcodes: pre-commit auto fixes

edescalona committed:
[MIG] stock_barcodes: Migration to 17.0

[17.0][IMP] stock_barcodes: Removing backorder action before validate, since it is included in the _pre_action_done_hook method

[17.0][IMP] stock_barcodes: Update readme. Changing partner design in scan view.

[17.0][FIX] stock_barcodes: Correcting the partial delivery process

[17.0][FIX] stock_barcodes: Tests

[IMP] stock_barcodes: Viewing current stock in physical inventory

[IMP] stock_barcodes: Change the order of the picking views. Allow capturing an action when validating the picking.

edescalona commented:
Ready for review.

mariano-aferrari commented:
Hi, not sure if this is the right place to put comments. But I've tested the module and I'm not able to validate a picking, even though the demand is fulfilled, it always warns the user to create a Partial delivery and even when accepting the partial delivery the picking is not validated. I've tried it manually and via the barcode interface and neither work. After removing the module it works again.

edescalona commented:
Hi @mariano-phi , I'll review the case you mentioned, but I should be able to validate it. There were also some changes that need to be uploaded. Thanks for your feedback.

vehi-invitu commented:
> Hi, not sure if this is the right place to put comments. But I've tested the module and I'm not able to validate a picking, even though the demand is fulfilled, it always warns the user to create a Partial delivery and even when accepting the partial delivery the picking is not validated. I've tried it manually and via the barcode interface and neither work. After removing the module it works again.

I have the same issue. I tested it on Runboat and on the picking Barcodes interface, when a product is scanned, the quantity is not updated in the received products. @edescalona Could you set your PR to draft while you're still working on it ? Thanks

mariano-aferrari commented:
Hi! @edescalona, do you have any update about a new version of the module? Thanks!

christian-ramos-tecnativa commented on stock_barcodes/models/stock_picking.py:
```diff
@@ -0,0 +1,73 @@
+                StockPicking, self.with_context(skip_backorder=True)
+            ).button_validate()
+        else:
+            pickings_to_backorder = self._check_backorder()
+            if pickings_to_backorder:
+                return pickings_to_backorder._action_generate_backorder_wizard(
+                    show_transfers=self._should_show_transfers()
+                )
```
Hello @edescalona I also encounter the same issue as @mariano-phi and @vehi-invitu, it looks like on this version there is a new field at `move_line` called `picked`. In order to be able to valdiate the picking and stop the backorder wizard to pop up this block should be removed as this is already handled by the `button_validate` function. You can find more info here [_pre_action_done_hook](https://github.com/odoo/odoo/blob/4e6bee49e98b055e5aebe89fb19ab6317003b682/addons/stock/models/stock_picking.py#L1193)
```suggestion
```

vehi-invitu replied:
Hello @Christian-RB, I tested these changes, but it’s not working on my side.

edescalona commented:
Hi @Christian-RB @mariano-phi @vehi-invitu, backorder issue resolved. If you find anything else in your reviews, please feel free to comment. Thanks for your feedback.

vehi-invitu commented:
> Hi @Christian-RB @mariano-phi @vehi-invitu, backorder issue resolved. If you find anything else in your reviews, please feel free to comment. Thanks for your feedback.

Hi @edescalona, thanks for the update. I tested your last commit (e22d607) on Runboat, but on my side, it's not working. I've attached a video recording to show the issues I encountered:
1. scanned 3 times but no increment,
2. entered quantity = 2 manually and confirmed, but no increment
3. clicked the +5 button and no increment.

It's not only about the backorder but also with the interface increment (when scanning barcode and entering the quantity manually).

Thanks for looking into this!
[Capture vidéo du 2025-06-20 14-16-29.webm](https://github.com/user-attachments/assets/eabd9c42-9b8d-4cce-b39a-a65fca5ddec0)

edescalona commented:
Hi @vehi-invitu thanks for your comments. I'll check what you're telling me and let you know.

edescalona commented:
Hello @vehi-invitu @mariano-phi @Christian-RB , if you could check the latest changes, thank you very much.

vehi-invitu commented:
> Hello @vehi-invitu @mariano-phi @Christian-RB , if you could check the latest changes, thank you very much.

Hello @edescalona, thank you for your changes. I tested the latest commit 2bb5dc3 and here are the tests I ran:
- **(WH/IN/00013)** Scanned 3 times -> but quantity stays at 1 (expected behavior?)
- Set quantity = 5 manually + Validate -> OK, a backorder **(WH/IN/00015)** is created for the remaining quantity
- Used the "+(`product_uom_qty`)" button + Validate -> OK
- **(WH/IN/00014)** Set `quantity` > `product_uom_qty` + Confirm -> Error: "Quantities scanned are higher than necessary" -> Confirm again (no error) + Validate = OK, but `quantity` > `product_uom_qty` ?

[Capture vidéo du 2025-07-07 10-42-30.webm](https://github.com/user-attachments/assets/ab258495-9f10-41de-8e58-6d054888540d)

Thanks again!

edescalona commented:
1. The latest commit allows for an action to be displayed during picking validation, for example, in the case of an expired product, as shown in the image.

![image](https://github.com/user-attachments/assets/262aa886-fd18-4894-acb5-696edbdf1aff)

2. The order of the view types (tree, kanban -> kanban, tree) in the picking view was also changed, so that regardless of where the view is accessed, the kanban is displayed as the priority.

![image](https://github.com/user-attachments/assets/04912640-18d1-465b-8fa0-2fe391196526)

edescalona commented:
Hi @vehi-invitu , regarding point 1:
Do you have the following option selected in the barcode option shown in the image?

![image](https://github.com/user-attachments/assets/4d0bab15-93ea-4694-b4f4-daf690a5dc96)

Regarding point 4:
Yes, in the default behavior, quantity can be greater than product_uom_qty

vehi-invitu commented:
Hello @edescalona, oops I forgot to select this option. I just tested it again with the option checked, but nothing happens when I scan the product.

[Capture vidéo du 2025-07-08 11-32-31.webm](https://github.com/user-attachments/assets/09ace917-81e3-42a6-8e68-8b9abbca1987)

edescalona commented:
Hi @vehi-invitu , I'll check it out and let you know. Thank you very much for your details.

mariano-aferrari commented:
Hello, any idea when this module could be completed and ready to use? Thanks @edescalona

edescalona commented:
> Hello, any idea when this module could be completed and ready to use? Thanks @edescalona

Hi @mariano-aferrari , it should be ready between this week and the beginning of the next, thanks.

mariano-aferrari commented:
Hello @edescalona any update? I really need this modulo for my daily operation. Thanks

edescalona commented:
Hi @mariano-aferrari sorry I haven't been able to update, I'll try to finish it as soon as possible, thanks.

edescalona committed:
[IMP] stock_barcodes: Defining field for partial delivery configuration

edescalona commented:
The commit (https://github.com/OCA/stock-logistics-barcode/pull/677/commits/17185ab0d83e7da76e1c15e43dffbddf59f23cc1), a new field is defined in the barcode options so that the user can enable partial delivery manually (using a wizard) or directly in the transfers on the barcode screen.

Note that the barcode option must be associated with the transaction type.

<img width="245" height="48" alt="image" src="https://github.com/user-attachments/assets/5e0d255c-df2a-40b7-8fc6-05edc906c74d" />

edescalona committed:
[IMP] stock_barcodes: Adding option to show actual amount (on hand) in barcode view.

edescalona commented:
The commit: https://github.com/OCA/stock-logistics-barcode/pull/677/commits/d2e270f3b9bdeb0c4a9e095ac30b5c77341c7d55

New option to display the actual quantity of the product.

<img width="245" height="48" alt="image" src="https://github.com/user-attachments/assets/7b2cc5da-cc4a-40ba-b01e-3facd044cc93" />

mariano-aferrari commented:
Thank you @edescalona. I've been doing some testing and I'm not sure if I'm making a mistake. If I configure an operation with a barcode Operation, for example, in receptions, PICKING IN barcode operation. When I select the picking, there's no product list, nevertheless if I don't configure the operation and select it manually, the product list is displayed perfectly. Am I missing anything?
<img width="879" height="256" alt="image" src="https://github.com/user-attachments/assets/0c4acd52-09e8-4aff-9fb9-371eaaf9a1c2" />
<img width="1703" height="1193" alt="image" src="https://github.com/user-attachments/assets/a608455e-139a-4fbd-bad4-e2b1894ba6b8" />
<img width="1334" height="830" alt="image" src="https://github.com/user-attachments/assets/3c2c6f01-8fcd-4eb4-991c-3fdcf45aadf2" />

edescalona commented:
Hi @mariano-aferrari , uncheck the Show pending moves option and it should now be displayed.

I'll explain that with this option, the lines where the demand and quantity are different will be displayed, and in v17, it will automatically add the same quantities. Please try it and let me know. Thanks.


<img width="597" height="57" alt="image" src="https://github.com/user-attachments/assets/3e23f381-8758-4b2d-a7ee-08e32a664582" />

edescalona committed:
[IMP] stock_barcodes: Adding option to display owner when performing an inventory adjustment.

edescalona commented:
The commit (https://github.com/OCA/stock-logistics-barcode/pull/677/commits/f099987c1ea581d1be40100ce966a0bced104c14) enables a new barcode option setting to allow owners to be selected in inventory settings.


<img width="224" height="54" alt="image" src="https://github.com/user-attachments/assets/10368086-2508-42d3-a31f-c11bc11a6e2a" />

edescalona committed:
[IMP] stock_barcodes: Identifying quantity fields for stock and packages.

edescalona commented:
The commit (https://github.com/OCA/stock-logistics-barcode/pull/677/commits/9014b36675e556e802caeeb744faee8a32503fc4)

<img width="885" height="250" alt="image" src="https://github.com/user-attachments/assets/9234a54e-0937-44dd-a2dd-ee0ae619bddd" />

edescalona committed:
[FIX] stock_barcodes: Test

edescalona committed:
[IMP] stock_barcodes: Change icon in adjustment inventory

pedrobaeza commented:
/ocabot migration stock_barcodes

@edescalona are you going to bring all the fixes and improvements done on 16 here?

edescalona commented:
Hi @pedrobaeza , this PR already had the improvements from the previous one and some other changes that we added in this version 17.

pedrobaeza commented:
Can you enumerate those changes for focusing in them? @Christian-RB you have a requested changes comment.

edescalona commented:
@pedrobaeza  Here's a list of the changes we've made in version 17:

All items are configurable in the barcode actions associated with the view in which the desired functionality is performed:

1. When making inventory adjustments, you should be able to view the current quantity while recording, taking into account the warehouse and owner. (Option: Show stock)

<img width="245" height="48" alt="image" src="https://github.com/user-attachments/assets/a9fc72f4-cf8f-467d-ac65-b3d270b4b7da" />


2. Be able to associate the stock owner from the barcode view. (Option: Show owner)

<img width="224" height="54" alt="image" src="https://github.com/user-attachments/assets/b3afa7df-bd18-4436-ab78-ce8520e3caf6" />

3. Partial delivery of pickings. (Option: Allow partial delivery)

<img width="245" height="48" alt="image" src="https://github.com/user-attachments/assets/ab0df476-31b8-409d-9bad-c7c0c13ba61f" />


4. In inventory adjustments, the actual quantity, the expected quantity, and the difference should be displayed.

<img width="1891" height="161" alt="image" src="https://github.com/user-attachments/assets/5c4be037-eeaf-469a-b456-e06192e4e425" />

5. Differentiate the quantity fields for stock and package.

<img width="885" height="250" alt="image" src="https://github.com/user-attachments/assets/39e6dcfc-f7c6-42c9-8d0e-9fb70a3b0c51" />

christian-ramos-tecnativa reviewed (approved):
Functional review OK

nurinayati commented:
Hello @edescalona I found some issues here:
1. When I try to scan the same product, it does not increase the quantity and remains 1
<img width="2873" height="940" alt="27 08 2025_17 37 03_REC" src="https://github.com/user-attachments/assets/8d018e98-6059-48f0-8417-102f5a7a1506" />

2. I am not sure how to use the barcode action scan. I have filled the barcode on the barcode action however, when I scan it, nothing happens. let me know if I missed some configuration :)
<img width="2876" height="372" alt="27 08 2025_17 38 23_REC" src="https://github.com/user-attachments/assets/29a2a3fa-1466-49d7-98c7-d2c743544425" />

3. By Default, the scanned qty should be 0, same logic as v16
<img width="2880" height="540" alt="27 08 2025_17 47 55_REC" src="https://github.com/user-attachments/assets/b2e91fa1-f1e7-4b75-a30f-1fb1f4a30170" />

edescalona commented:
Hi @nurinayati , regarding the points you mentioned:
1. Go to the Inventory -> Configuration -> Barcode Options menu and select the Accumulate Read Quantity option, as shown in the image.

<img width="1121" height="478" alt="image" src="https://github.com/user-attachments/assets/d3fcec40-6720-43fa-a8d6-767a0371ed32" />

2. That barcode field is for opening the barcode action from the main Barcodes view according to the code reflected in the field that is shown to me.

<img width="1532" height="416" alt="image" src="https://github.com/user-attachments/assets/53054eb6-9f8e-462d-9476-ffb14e1d8621" />

3. By default, in version 17, demand and quantity are equal.


Thank you for your comments.

christian-ramos-tecnativa commented:
Hello @edescalona,

After some testing, I noticed that in an internal transfer, the destination location is filtered by the current warehouse:
<img width="1097" height="465" alt="image" src="https://github.com/user-attachments/assets/a7fd3cd4-ef78-44d0-b5da-987b7a5ada12" />

While I understand that internal transfers are generally intended for moves between locations within the same warehouse, the standard behavior actually allows selecting locations from other warehouses:
<img width="1221" height="694" alt="image" src="https://github.com/user-attachments/assets/0b0d2207-da7f-4631-94d0-abd75f6adca8" />

What do you think about keeping the filter consistent with the standard?

edescalona commented:
Hi @Christian-RB , it could be an option within the barcode options, leaving both approaches.

What do you think, @pedr @ws, about what was discussed (https://github.com/OCA/stock-logistics-barcode/pull/677#issuecomment-3228868678)?

nurinayati commented:
Thank you for your reply @edescalona :)
1. I have checked the boolean, and the issue is still there.
2. Yes I was trying to scan the barcode on that screen and nothing happens. please let me know if i missed anything to check
3. Ok, that makes sense, but I think it would be better to display the unscanned quantity like Barcode app in Odoo EE 17
4. I found another error when trying to scan a barcode that is not in the list. it should have an error-handling message. How to reproduce: display 2 picking with different tabs of barcode app and scan a product that is not in the list
https://www.awesomescreenshot.com/video/43597551?key=afffe58a592f8eacf79ceab0769be8fa
<img width="2841" height="1058" alt="28 08 2025_10 31 35_REC" src="https://github.com/user-attachments/assets/96338b9b-887a-4bca-a45e-5909440effa6" />

edescalona commented:
Hi @nurinayati , thanks for your comments. Could you please copy the error message you showed me in point 4? I'd like to follow the steps you mentioned and it doesn't show me that error. Thanks.

nurinayati commented:
Hi @edescalona Please see the error below. yes the error is not always showing, but I got the error somehow if I have 2 active barcode screen with different stock pickings :)

UncaughtClientError > TypeError
Uncaught Javascript Error > Cannot read properties of undefined (reading 'writeText')

TypeError: Cannot read properties of undefined (reading 'writeText')
    at ClientErrorDialog.onClickClipboard (http://oca-stock-logistics-barcode-17-0-pr677-281c21440047.runboat.odoo-community.org/web/assets/dbbc27e/web.assets_web.min.js:2382:48)
    at Object.mainEventHandler (http://oca-stock-logistics-barcode-17-0-pr677-281c21440047.runboat.odoo-community.org/web/assets/dbbc27e/web.assets_web.min.js:1593:77)
    at HTMLButtonElement.listener (http://oca-stock-logistics-barcode-17-0-pr677-281c21440047.runboat.odoo-community.org/web/assets/dbbc27e/web.assets_web.min.js:748:15)

florian-dacosta commented:
Hello @edescalona,
Thanks for this migration work.

> 3\. By default, in version 17, demand and quantity are equal.

As @nurinayati I concerned about it.
The first goal of the barcode reader is to scan the products than need to be picked. If the barcode reader application already fill the quantity as if it had been picked, it make the application useless in most of the "picking" cases no ?
I don't understand how to use the barcode reader application for picking if the quantity are prefilled !

Odoo SA did remove the qty_done field on `stock.move.line` from the `stock` module, but they still have the same behavior on their barcode app (quantity of each line are set to 0 and increased on scan). I guess they did add it or a similar field in the barcode module.
Also I see you did add the qty_done field back on `stock.move.line`, I think we could/should use it along with the `picked` boolean (else I am not sure why you had to add it back?). 
This way, as in v16, we can start with a 0 quantity, record what has been scanned with qty done, and  use the picked boolean when the full quantity of the line is scanned.

From what I just did test on a v18 runbot, the enterprise application does split the stock.move.line in case everything is not scanned.
For example, I need to pick a product 40 times, I have 1 `stock.move.line`. If I scan the barcode 10 times and then leave the barcode interface, we will have then 2 `stock.move.line`. One with quantity 10 marked as `picked`, the other of 30, which remains to do.


A second thing I saw during the quick test I made on runboat : when I edit the quantity of a line manually, I can't set a 0 quantity. When I put 0 and then confirm, nothing happens.
<img width="1850" height="929" alt="image" src="https://github.com/user-attachments/assets/07c8a3e6-790e-4df5-b36e-14d288446069" />

I'd appreciate your feedback on those 2 points!
Thanks again for your huge work on this module.

etobella committed:
[IMP] stock_barcodes: Remove unnecessary dependency

etobella commented:
Before doing anything, review this:

https://github.com/OCA/stock-logistics-barcode/pull/718

Commit:
[IMP] stock_barcodes: Use qty_picked from stock_move_line_qty_picked

Commit:
[DON'T MERGE] test-requirements.txt

christian-ramos-tecnativa commented:
This PR now depends on [OCA/stock-logistics-workflow#2132](https://github.com/OCA/stock-logistics-workflow/pull/2132)
Thanks, @florian-dacosta, for your guidance! 🙏
The functionality for splitting `stock.move.line` is still missing.
@sergio-teruel, what do you think about setting up a short meeting to discuss the remaining work for versions 17 and 18?
It seems we might be working on the same issues in parallel, so coordinating could help us split the workload more efficiently.
Thanks!

pedrobaeza commented:
Please check the work done on the 18.0 migration in #725 to coordinate.

psc committed:
Translated using Weblate (German)

Currently translated at 86.3% (19 of 22 strings)

Translation: stock-logistics-barcode-17.0/stock-logistics-barcode-17.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-17-0/stock-logistics-barcode-17-0-barcodes_generator_abstract/de/

psc committed:
Translated using Weblate (German)

Currently translated at 95.4% (21 of 22 strings)

Translation: stock-logistics-barcode-17.0/stock-logistics-barcode-17.0-barcodes_generator_abstract
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-17-0/stock-logistics-barcode-17-0-barcodes_generator_abstract/de/

psc committed:
Translated using Weblate (German)

Currently translated at 100.0% (18 of 18 strings)

Translation: stock-logistics-barcode-17.0/stock-logistics-barcode-17.0-product_multi_barcode
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-17-0/stock-logistics-barcode-17-0-product_multi_barcode/de/

psc committed:
Added translation using Weblate (German)

Commit:
Translated using Weblate (German)

Currently translated at 100.0% (17 of 17 strings)

Translation: stock-logistics-barcode-17.0/stock-logistics-barcode-17.0-barcodes_generator_product
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-17-0/stock-logistics-barcode-17-0-barcodes_generator_product/de/

mariano-aferrari commented:
Hi everyone, any progress on this module for Odoo 17?. Could you please provide an update on where you are standing with it?. Thanks!

giarve committed:
[FIX] stock_barcodes: fix opening client action permissions

pedrobaeza commented:
You should take into account #733

Commit:
Merge branch '17.0-mig-stock_barcodes' of https://github.com/BinhexTeam/stock-logistics-barcode into 17.0-mig-stock_barcodes

mariano-aferrari commented:
Hi, is this module production ready?, can you comment on the progress please?

mariano-aferrari commented:
@edescalona @pedrobaeza Can you share what's the status of this migration?. Thanks!

cvinh commented:
> @edescalona @pedrobaeza Can you share what's the status of this migration?. Thanks!

Hello @mariano-aferrari feel free to test and review this migration. When it will have at least 2 approvals it could be merged
